autogluon/utils/tabular/features/auto_ml_feature_generator.py

```diff
@@ -146,33 +144,37 @@ def generate_nlp_ngrams(self, X, features_nlp_current, downsample_ratio: int = N
         for i, nlp_feature in enumerate(features_nlp_current):
             vectorizer_fit = self.vectorizers[i]
 
-            transform_matrix = vectorizer_fit.transform(X[nlp_feature].values)
+            if nlp_feature == '__nlp__':
+                text_data = ['. '.join(row) for row in X[self.features_nlp].values]
+            else:
+                text_data = X[nlp_feature].values
+            transform_matrix = vectorizer_fit.transform(text_data)
 
-            predicted_ngrams_memory_usage_bytes = len(X) * 8 * (transform_matrix.shape[1] + 1) + 80
-            mem_avail = psutil.virtual_memory().available
-            mem_rss = psutil.Process().memory_info().rss
-            # TODO: 0.25 causes OOM error with 72 GB ram on nyc-wendykan-lending-club-loan-data, fails on NN or Catboost, distributed.worker spams logs with memory warnings
-            # TODO: 0.20 causes OOM error with 64 GB ram on NN with several datasets. LightGBM and CatBoost succeed
-            max_memory_percentage = 0.15  # TODO: Finetune this, or find a better metric
-            predicted_rss = mem_rss + predicted_ngrams_memory_usage_bytes
-            predicted_percentage = predicted_rss / mem_avail
             if not self.fit:
+                predicted_ngrams_memory_usage_bytes = len(X) * 8 * (transform_matrix.shape[1] + 1) + 80
+                mem_avail = psutil.virtual_memory().available
+                mem_rss = psutil.Process().memory_info().rss
+                # TODO: 0.25 causes OOM error with 72 GB ram on nyc-wendykan-lending-club-loan-data, fails on NN or Catboost, distributed.worker spams logs with memory warnings
+                # TODO: 0.20 causes OOM error with 64 GB ram on NN with several datasets. LightGBM and CatBoost succeed
+                max_memory_percentage = 0.15  # TODO: Finetune this, or find a better metric
```

Comment:
does this value of 0.15 now need to be tweaked since model may convert the data out of uint8?

Reply:
This value is calculated as if the ngrams are taking up int64, as prior, so the amount of ngrams generated is identical as previous, despite taking 1/8th the memory.